flink-libraries/flink-table/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java

```diff
@@ -719,7 +724,13 @@ public static long timestampStringToUnixDate(String s) {
 	}
 
 	public static long unixDateExtract(TimeUnitRange range, long date) {
-		return julianExtract(range, (int) date + EPOCH_JULIAN);
+		switch (range) {
+			case EPOCH:
+				// no need to extract year/month/day, just multiply
+				return date * SECONDS_PER_DAY;
+			default:
+				return julianExtract(range, (int) date + EPOCH_JULIAN);
+		}
 	}
 
 	private static int julianExtract(TimeUnitRange range, int julian) {
```

Comment:
I'm a little bit worried about the `julian` thing in this method name... I have a bad feeling that it supposed to be Gregorian calendar here...

Reply:
This class should not be changed. It is copied from Calcite until a certain issue is fixed. If we apply changes here, then just to update it to the latest Calcite code base.